src/protocol/notarization/encryption.md

```diff
@@ -0,0 +1,143 @@
+
+The equality check at the end of our protocol ensures that premature leakage is detected with a probability of $1 - 2^{-k}$ where k is the number of leaked bits. The Notary is virtually guaranteed to detect significant leakage and can abort prior to notarization.
+
+### Plaintext Leakage
+
+Our protocol assures _no leakage_ of the plaintext to the Notary during both encryption and decryption. The Notary reveals their keyshare at the end of the protocol, which allows the Notary to open their garbled circuits and oblivious transfers completely to the User. The User can then perform a series of consistency checks to ensure that the Notary behaved honestly. Because these consistency checks do not depend on any inputs of the User, aborting does not reveal any sensitive information (in contrast to standard DualEx which does).
+
+### Integrity
+
+During the entirety of the TLS session the User performs the role of the garbled circuit generator, thus ensuring that a malicious Notary can not corrupt or otherwise compromise the integrity of messages sent to/from the Server.
+
+There is one exception to the above, and that is during the transmission of the first encrypted handshake messages in which The Notary has the opportunity to pick a different value for their keyshare. The result of this would simply be a corrupt payload. Either the User or Server would detect this immediately and subsequently abort the connection prior to the transmission of any application data. Past this stage the Notary is committed to using the same keyshare for the rest of the session and thus has no opportunity to introduce a malicious input.
+
+### Notation
+
+* $p$ is one block of plaintext
+* $c$ is the corresponding block of ciphertext, ie $c = \mathsf{Enc}(k, ctr) \oplus p$
+* $k$ is the cipher key
+* $k_1$ and $k_2$ are the User's and Notary's keyshares, respectively. That is, $k = k_1 \oplus k_2$
+* $z$ is a mask randomly selected by the User
+* $ectr$ is the encrypted counter-block, ie $ectr = \mathsf{Enc}(k, ctr)$
+* $G$ denotes a garbled circuit
+* $d$ denotes output decoding information where $\mathsf{De}(d, [v]) = v$
+* $\mathsf{Enc}$ denotes the block cipher used by the TLS session
+* $\mathsf{PRG}$ denotes a secure pseudorandom generator
+* $\mathsf{com}_x$ denotes a binding commitment to the value $x$
+* $[x]_A$ denotes a garbled encoding of $x$ chosen by party $A$
+
+## Encryption Protocol
+
+### Phase 1: 2PC
+
+1. The User creates a garbled circuit $G_U$ which computes $c = ectr \oplus p$ and generates output label commitments $\mathsf{com}_{[c]_U}$. She sends $G_U$, $[k_1]_U$, $[p]_U$, $d_U$ and $\mathsf{com}_{[c]_U}$ to the Notary.
```

Comment:
The things below repeat the decsription of DEAP. Is there a way to cast it so that we can  just say: "the parties run DEAP with Notary's inputs being X and User's inputs being Y"?